src/eigen/preconditioners.jl

```diff
@@ -11,6 +11,19 @@
 import LinearAlgebra: ldiv!
 import LinearAlgebra: mul!
 
+# Returns dot(A[:, i], Diagonal(P.kin), B[:, i])), for all columns of A, B
+@views function diag_prod(A::AbstractArray{T}, B::AbstractArray{T}; diag=nothing) where {T}
```

Comment:
Maybe we should move all of these helper functions to a single file in `src/common`. Because some being in `preconditioners.jl` and some in LOBPCG does not really help finding them in the future.

Reply:
Maybe in a new file `src/eigen/common.jl` ? This would keep the locality of functions needed by LOBPCG for the moment when it branches out of DFTK.